cmd/globals.go

```diff
@@ -123,8 +123,8 @@ var (
 	globalHTTPServerErrorCh = make(chan error)
 	globalOSSignalCh        = make(chan os.Signal, 1)
 
-	// Enable HTTP request/response headers and body logging.
-	globalHTTPTrace bool
+	// File to log HTTP request/response headers and body.
+	globalHTTPTraceFile *os.File
```

Comment:
Is `globalHTTPTraceFile` goroutine safe? or write to this file always happens serially?

Reply:
I don't think that is.. they might get interspersed. So this is my question as well.

Reply:
I tried few tests on Ubuntu 17.04 and was unable to make the data interleave across multiple writes in different goroutines. 
As far as Windows is concerned, internally `WriteFile` API is called, which is guaranteed to be atmoic. As far as other OSes are concerned, according to https://stackoverflow.com/questions/29981050/concurrent-writing-to-a-file relatively newer versions of the kernel should support this assumption.
I can create a wrapper around the `Write` function to make sure that there will not be any interleaving of data. Let me know what you guys think.

Reply:
I am okay if this investigation is true.

Reply:
it's taken care of:
```
                // Write the contents in one shot so that logs don't get interspersed.
                output.Write(b.Bytes())
```

Reply:
LGTM